interfaces.go

```diff
@@ -21,9 +23,15 @@ type collectorResponse interface {
 	Disable() bool
 }
 
+type reportRequest struct {
+	thriftRequest *lightstep_thrift.ReportRequest
+	grpcRequest   *cpb.ReportRequest
```

Comment:
nit: this is actually a `protoRequest` not a `grpcRequest`. In the `v0.15.0` branch, I am adding proto over http and it uses the same request type.